lib/charms/operator_libs_linux/v0/juju_systemd_notices.py

```diff
@@ -159,7 +176,7 @@ class ServiceStoppedEvent(EventBase):
 class SystemdNotices:
     """Observe systemd services on your machine base."""
 
-    def __init__(self, charm: CharmBase, services: List[str]) -> None:
+    def __init__(self, charm: CharmBase, *services: Service) -> None:
```

Comment:
I don't think we should make a breaking change for this relatively minor improvement. It's easy in Python to handle both dynamically. A couple of possibilities:

1) Make the type `List[str | Service]` or similar.
2) Make it so if the first argument is a list, it treats it as a list of str, otherwise *args.
2) Keep it `List[str]`, and if a string has an `=` in it, use the right hand side as an alias. Something like `SystemdNotices(['snap.slurm.slurmd=slurmd', 'noalias'])`

Reply:
I did think about going with option 1 after changing the function signature to `*services: Service`. It's my preferred option compared to the other 2. 

Key thing to note is that we'll need to set the signature to `List[Union[str, Service]]`. Support for using `|` to represent a union type annotation [wasn't added until Python 3.10+](https://docs.python.org/3/library/typing.html#typing.Union).

Reply:
Reverted to compatible API so we only have to bump the LIBPATCH version.